src/DDTrace/Configuration/EnvVariableRegistry.php

```diff
@@ -35,7 +40,15 @@ public function __construct($prefix = 'DD_')
      */
     protected function get($key)
     {
-        $value = getenv($this->convertKeyToEnvVariableName($key));
+        if (array_key_exists($key, $this->keyTranslator)) {
+            $key = $this->keyTranslator[$key];
+        } else {
+            $new_key = $this->convertKeyToEnvVariableName($key);
+            $this->keyTranslator[$key] = $new_key;
+            $key = $new_key;
+        }
+
+        $value = getenv($key);
```

Comment:
Agreed its much cleaner . Seems my performance hat squeezed my head maybe a little too tight :)
And I wrote it like that to avoid extra array lookup :D.
However it does look like my version produces 5 more opcodes to process. I think going with your approach with one extra array lookup is less expensive than processing more opcodes. And cleaner PHP too :)

Reply:
Actually Just noticed this key translation is actually cached already :) by all the (bool|string|...l)Value thats actually used to access the values. 

Probably the only way to speed it up would be to use hashmap thats shared between requests.